api/app/controllers/api/v1/offers_controller.rb

```diff
@@ -1,31 +1,396 @@
+        if params[:update]
+          update_print_status(offer_id)
         end
+        offer = Offer.find(offer_id)
+        offers.push(offer)
+      end
+      # create a pdf concatenation of all the offers using CombinePDF
+      out_pdf = CombinePDF.new
+      offers.each do |offer|
+        rendered = render_to_string pdf: "contract", inline: get_contract_html(offer, true), encoding: "UTF-8"
+        out_pdf << CombinePDF.parse(rendered)
+      end
+      send_data out_pdf.to_pdf, filename: "contracts.pdf", disposition: "inline"
+    end
+
+    def get_contract
+      rendered = get_contract_html(Offer.find(params[:offer_id]))
+      render pdf: "contract", inline: rendered
+    end
+
+    def get_contract_student
+      rendered = get_contract_html(Offer.find(params[:offer_id]))
+      render pdf: "contract", inline: rendered
+    end
+
+    def get_contract_student_html
+      rendered = get_contract_html(Offer.find(params[:offer_id]))
+      render inline: rendered
+    end
+
+    def set_status_student
+      if params[:status] == "accept" || params[:status]== "reject"
+        status_setter({offer_id: params[:offer_id], status: params[:status]})
+      else
+        render status: 404, json: {success: false, message: "Error: no permission to set such status"}
+      end
+    end
+
+    '''
+      Sets the offer status as either `Accepted`, `Rejected`, or `Withdrawn`. This
+      action is for the admin.
+    '''
+    def set_status
+      status_setter(params)
+    end
 
-        # POST /email_offer
-        def email_offer
-            # TODO
-            render_error('')
+    def clear_hris_status
+      params[:offers].each do |id|
+        offer = Offer.find(id)
+        offer.update_attributes!(hr_status: nil, print_time: nil)
+      end
+    end
+
+    def accept_offer
+      offer = Offer.find(params[:offer_id])
+      if offer[:status]!="Unsent"
+        offer.update_attributes!(status: "Accepted", accept_date: DateTime.now)
+        offer = offer.format
+        render status: 200, json: {message: "You've changed the status of #{offer[:applicant][:first_name]} #{offer[:applicant][:last_name]}'s offer for #{offer[:position]} to 'Accepted'."}
+      else
+        render status: 404, json: {message: "Error: You can't accept an Unsent offer."}
+      end
+    end
+
+    def reset_offer
+      offer = Offer.find(params[:offer_id])
+      offer.update_attributes!(status: "Unsent", accept_date: nil, commentary: nil, hr_status: nil, nag_count: nil,
+                               print_time: nil, send_date: nil, signature: nil)
+      offer = offer.format
+      render status: 200, json: {message: "You've changed the status of #{offer[:applicant][:first_name]} #{offer[:applicant][:last_name]}'s offer for #{offer[:position]} to 'Unsent'."}
+    end
+
+    private
+    def get_contract_html(offer, office_template=false)
+      contract_dir = "#{Rails.root}/app/views/contracts/#{ENV["CONTRACT_SUBDIR"]}"
```

Comment:
this should be an `to_html` method for the Offer model instead of a controller method.